### PR TITLE
[bug 1087571] Codify editor settings using editorconfig

### PR DESCRIPTION
**Pull request by lgp171188**
Add the settings currently used by kitsune. Can add more specific settings later if required.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# http://editorconfig.org
+
+root = true
+
+[*]
+indent_style = space
+indent_size = 4
```
Comment (willkg) [inline .editorconfig]: So, this is where it gets tricky. We use 4-space indents for .py and .js files. For .less files we use 2-space indents. I don't know offhand whether there are other rules, but I think we should move the indent bits to a `[*.py,*.js]` block and add this:

```
[*.less]
indent_style = space
indent_size = 2
```

**Comment by willkg** [comment]
Other than that thing about `indent_size`, this looks great.

We probably want to mention something in the docs so that contributors can set up editorconfig things in their editor of choice. It's probably sufficient to put it in the Getting Started guide and link to the part of the editorconfig site where they mention editor plugins.

This looks great! Thank you!

**Comment by lgp171188** [comment]
Implemented the changes suggested. Let me know if it is okay @willkg

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+[*]
+insert_final_newline = true
+trim_trailing_whitespace = true
+end_of_line = lf
+charset = utf-8
+
+[*.{js,py}]
+indent_style = space
+indent_size = 4
+
```
Comment (willkg) [inline .editorconfig]: This needs one more section:

```
[*.less]
indent_style = space
indent_size = 2
```

Comment (lgp171188) [inline .editorconfig]: Forgot to add it in my previous commit. Added it now

**Comment by willkg** [comment]
Looks great! Thank you!

**Pull request merged**